rfd/01-jobby.md

```diff
@@ -0,0 +1,175 @@
+    // Stream starts at the beginning of the process's stdout
+    func (j *Job) Stdout() io.ReadCloser
+
+    // Returns a ReadCloser providing access to the job's standard output stream
+    // Stream starts at the beginning of the process's stderr
+    func (j *Job) Stderr() io.ReadCloser
+```
+
+#### Note on streaming output
+Stdout/Stderr will be available even after the job completes. Each stream will be persisted to a file (created by the job package at creation time) at a path specified by the caller (either concurrent with process execution or perhaps flushed upon job completion). After the job exits, the lifecycle of these output files is up to the caller. They can delete them later or even just direct the job output to an ephemeral filesytem (tmpfs).
+
+I've chosen to return a ReadCloser from these functions to provide caller with the opportunity to "quit/detach" rather than block indefinitely while waiting for output. This will hopefully provide simple and intuitive interface for callers and make the server implementation a breeze. This is probably the most difficult sub-problem of the project. For scenarios like this, I would much rather manage higher *internal complexity* to provide a simple interface, rather than define a complex interface that might shift complexity onto the caller. 
+
+#### Aside
+At first I considered adding some sort of "Manager" or "JobStore" type for assigning job identifiers and handling job lookup, but this is out of scope for the package. It's unlikely that other potential consumers of this package would get any use out of it, so that logic should be implemented elsewhere.
+
+
+### Server
+A server implementation will provide a gRPC service interface to the job package. Authentication is managed via mTLS, so each client is expected to present a valid cert. In a nutshell the server will:
+
+* Handle lookup and retrieval of jobs using a UUID assigned to each job
+  * A simple KV store in-memory will be used for lookup (a map). It will not persist upon a restart. 
+  * The server will asign each job a UUID as an identifier, as well as associate user and group information with each job. 
+    * UUID is preferred as an identifier over the process's PID since PIDs will eventually wrap.
+* Manage Authentication of users as well implement a simple access control policy for accessing jobs.
+  * Each job will be owned by the user who created it. **Only this user will be allowed to stop the job.**
+  * Each user *may* also belong to a single "group". Group members are allowed view the status and output of any job created by another user within their group.
+  * Should a user's group membership change, that user retains ownership of their job and the previous group retains access to status and output of any existing jobs associated to the group.
+  * User and Group will be determined by examining the **Distinguished Name** and **Organizational Unit** respectively as found in the client certificate.

```

Comment:
No need to implement group membership. Only showing the job to the user that created it is enough.

Hopefully this saves you some time on implementation.